src/Microsoft.SqlTools.ServiceLayer/EditData/EditTableMetadata.cs

```diff
@@ -0,0 +1,96 @@
+            Debug.Assert(dbColumns.Count == smoObject.Columns.Count);
+
+            // Create the columns for edit usage
+            columns = new List<EditColumnWrapper>();
+            for (int i = 0; i < dbColumns.Count; i++)
+            {
+                Column smoColumn = smoObject.Columns[i];
+                DbColumnWrapper dbColumn = dbColumns[i];
+
+                // A column is trustworthy for uniqueness if it can be updated or it has an identity
+                // property. If both of these are false (eg, timestamp) we can't trust it to uniquely
+                // identify a row in the table
+                bool isTrustworthyForUniqueness = dbColumn.IsUpdatable || smoColumn.Identity;
+
+                EditColumnWrapper column = new EditColumnWrapper
+                {
+                    DbColumn = dbColumn,
+                    Ordinal = i,
+                    EscapedName = SqlScriptFormatter.FormatIdentifier(dbColumn.ColumnName),
+                    IsTrustworthyForUniqueness = isTrustworthyForUniqueness,
+
+                    // A key column is determined by whether it is in the primary key and trustworthy
+                    IsKey = smoColumn.InPrimaryKey && isTrustworthyForUniqueness
+                };
+                columns.Add(column);
+            }
+
+            // Determine what the key columns are
+            keyColumns = columns.Where(c => c.IsKey).ToList();
+            if (keyColumns.Count == 0)
+            {
+                // We didn't find any explicit key columns. Instead, we'll use all columns that are
+                // trustworthy for uniqueness (usually all the columns)
+                keyColumns = columns.Where(c => c.IsTrustworthyForUniqueness).ToList();
+            }
+
+            // If a table is memory optimized it is Hekaton. If it's a view, then it can't be Hekaton
+            Table smoTable = smoObject as Table;
+            IsHekaton = smoTable != null && smoTable.IsMemoryOptimized;
+
+            // Escape the parts of the name
+            string[] objectNameParts = {smoObject.Schema, smoObject.Name};
+            EscapedMultipartName = SqlScriptFormatter.FormatMultipartIdentifier(objectNameParts);
+        }
+
+        /// <summary>
+        /// Read-only list of columns in the object being edited
+        /// </summary>
+        public IEnumerable<EditColumnWrapper> Columns => columns;
+
+        /// <summary>
+        /// Full escaped multipart identifier for the object being edited
+        /// </summary>
+        public string EscapedMultipartName { get; }
+
+        /// <summary>
+        /// Whether or not the object being edited is hekaton/memory optimized
+        /// </summary>
+        public bool IsHekaton { get; }
+
+        /// <summary>
+        /// Read-only list of columns that are used to uniquely identify a row
+        /// </summary>
+        public IEnumerable<EditColumnWrapper> KeyColumns => keyColumns;
```

Comment:
>keyColumns [](start = 60, length = 10)

Does this really return a ReadonlyCollection? Shouldn't you call keyColumns.AsReadOnly instead? And do the same for Columns?

Also, as mentioned before: the desire to shorten things is nice, but we should agree whether we really want to do this or keep with the boring (but clear) old { get { return xyz; } } syntax.